packages/admin/cms-admin/src/generator/future/generateGrid.ts

```diff
@@ -477,9 +480,27 @@ export function generateGrid(
                 .map((column) => {
                     const columnDefinition: TsCodeRecordToStringObject = {
                         field: `"${column.name.replace(/\./g, "_")}"`, // field-name is used for api-filter, and api nests with underscore
-                        headerName: `intl.formatMessage({ id: "${instanceGqlType}.${column.name}",  defaultMessage: "${
+                        renderHeader: `() => (
+                                    <div style={{ display: "flex", alignItems: "center" }}>
+                                        <Typography fontWeight={400} fontSize={14}>
+                                            {intl.formatMessage({ id: "${instanceGqlType}.${column.name}", defaultMessage: "${
                             column.headerName || camelCaseToHumanReadable(column.name)
-                        }" })`,
+                        }" })}
+                                        </Typography>
+                                        ${
+                                            column.tooltipMessage
+                                                ? `<Tooltip
+                                            trigger="hover"
+                                          title={<FormattedMessage id="${instanceGqlType}.${column.name}.tooltip" defaultMessage="${column.tooltipMessage}" />}
+                                        >
+                                            <IconButton>
```

Comment:
Should this have an `onClick` event? If not, this should probably not be a button.